homeassistant/components/soma/cover.py

```diff
@@ -21,77 +38,179 @@ async def async_setup_entry(
     """Set up the Soma cover platform."""
 
     devices = hass.data[DOMAIN][DEVICES]
+    entities = []
+
+    for device in devices:
+        # Assume a shade device if the type is not present in the api response (Connect <2.2.6)
+        if ("type" in device.keys()) and (device["type"].lower() == "tilt"):
+            entities.append(SomaTilt(device, hass.data[DOMAIN][API]))
+        else:
+            entities.append(SomaShade(device, hass.data[DOMAIN][API]))
+
+    async_add_entities(entities, True)
+
+
+class SomaTilt(SomaEntity, CoverEntity):
+    """Representation of a Soma Tilt device."""
+
+    @property
+    def device_class(self):
+        """Return the class of this device."""
+        return DEVICE_CLASS_BLIND
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        supported_features = (
+            SUPPORT_OPEN_TILT
+            | SUPPORT_CLOSE_TILT
+            | SUPPORT_STOP_TILT
+            | SUPPORT_SET_TILT_POSITION
+        )
+        return supported_features
+
+    @property
+    def current_cover_tilt_position(self):
+        """Return the current cover tilt position."""
+        return self.current_position
+
+    @property
+    def is_closed(self):
+        """Return if the cover tilt is closed."""
+        return self.current_position == 0
+
+    def close_cover_tilt(self, **kwargs):
```

Comment:
Please see comments in `stop_cover_tilt`

Reply:
Fixed